### PR TITLE
fix(gui-app): keep the composer menu on screen and clear of the keyboard

### PR DESCRIPTION
**Pull request by pranshugupta54**
Three fixes to the composer's picker menu (slash commands, `@` mentions) on a phone.

## 1. The menu stays on screen

**What you see.** The menu opens at the caret. With the caret toward the right of the screen, the menu ran off the side, cutting off command names and descriptions. It now stays fully on screen, 8px from either edge.

**Why.** The menu is clamped to the viewport after it is placed at the caret, but that clamp was only enabled vertically. Nothing pulled a wide menu back horizontally, so on a narrow screen it overflowed whichever side it was aligned to. The clamp is now on for both axes.

**Unchanged.**
- The menu still opens below the caret when there is room and above it otherwise.
- The vertical clamp is kept: a menu taller than the room on either side of the caret is still slid back on screen (over the caret line if needed) rather than left with its header or rows unreachable.
- A menu that already fits stays anchored at the caret.
- The preview panel shown beside the menu on wider screens (not on phones) still anchors to the highlighted row, so it moves with the menu. Its positioning is not changed here.

## 2. Tapping a menu row keeps the keyboard up

**What you see.** With the keyboard up and the menu open, tapping a command made the keyboard drop and immediately come back. The keyboard now stays up.

**Why.** On mobile, tapping anywhere outside the focused text field puts the keyboard away. The menu is rendered outside the editor's part of the page, so a tap on a row counted as "outside": the editor lost focus (keyboard down), then choosing the command put focus back (keyboard up). A text field's popups now declare themselves as belonging to it (`data-text-entry-popup`), and a tap inside one is treated like a tap inside the field. The composer menu (rows and its header buttons) and its preview panel carry the marker.

**Unchanged.**
- Tapping anywhere else, buttons and links included, still dismisses the keyboard.
- A hard downward flick or drag over the menu still dismisses the keyboard, as it does anywhere else; the menu stays open when that happens.
- Row selection itself (the commit and focus path) is untouched.

## 3. The menu opens where the keyboard is not

**What you see.** With the keyboard up, the menu opened below the caret and most of it sat behind the keyboard - only the header and two or three rows were visible, for the slash menu and the `@` menu alike. It now opens above the caret when the room below is keyboard, and a list too tall for the space left is capped to it and scrolls. It moves when the keyboard opens or closes while the menu is up.

**Why.** On iOS the keyboard is drawn over the app rather than shrinking it; the app's own layout ends above it by subtracting the keyboard's height. The menu floats over the whole screen instead, and measured its room against the full screen height, so it counted the keyboard's strip as free space. It now measures against the screen less that strip (and less the status-bar strip at the top), using the same keyboard height the layout uses. The preview panel beside the menu on wider screens uses the same area. The same area also keeps both the menu and the panel clear of the phone's other screen-edge strips - the camera housing at the sides in landscape, and the home-indicator strip at the bottom when the keyboard is closed.

**Unchanged.** On Android the system already shrinks the app for the keyboard, and in a browser or on desktop there is no overlaid keyboard; the keyboard height is 0 there, so placement is exactly as before.

## Tests

- `composer-menu-middleware.test.ts` runs the menu's real positioning middleware through floating-ui with a 390px-wide viewport: a menu opened near the right edge (above and below the caret) and at the left edge lands inside the viewport, a fitting menu stays at the caret, and an over-tall menu stays inside vertically. The edge cases fail against the previous configuration.
- `shell-gestures.test.tsx`: with a focused editor, a tap on an element inside a `data-text-entry-popup` surface rendered elsewhere in the page leaves the editor focused. Fails before the fix (the editor is blurred).
- `composer-menu.test.tsx`: the real rendered menu's rows and header, and its preview panel, are recognised as the editor's popup by the same helper the gesture recognizer uses. Fails with the marker removed.

- `composer-menu-middleware.test.ts` (keyboard): with a 336px keyboard on a 390x844 viewport, a menu at a caret just above the keyboard resolves above it and ends above the keyboard; the same caret with no keyboard still opens below; a list taller than the room above the keyboard is capped to it; the top status-bar strip is kept clear; the open-time side choice counts neither strip as room. These fail against the previous middleware.
- `composer-menu.test.tsx` / `mention-preview-panel.test.tsx`: the slash menu, the `@` root, an `@` sub-step with its Filter/Refresh bar, and the preview panel all reposition when the keyboard opens and again when it closes.

The keyboard-tap tests prove the recognizer no longer blurs the editor for a tap on the menu. Fixes 1 and 2 were checked by hand on an iOS Simulator with the software keyboard.

### CODE REVIEW

File: clients/gui-app/src/components/chat/composer/menu/mention-preview-panel-fit.ts [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+ *
+ * What that space is differs by axis, because the panel's `shift` clamps it
+ * horizontally but not vertically:
+ * - Width: with a horizontal shift enabled, `size` reports the whole
+ *   boundary's width less padding, NOT the room beside the anchor row. The
+ *   width check therefore only hides the panel on a viewport too narrow for
+ *   it; a panel with no room on either side of the menu is slid back on
+ *   screen, over the list, and still shows.
+ * - Height: measured from the anchor row's top to the boundary's bottom less
+ *   padding, so a row too close to the bottom edge hides the panel.
```
[the remark this answers is not in the document]
Comment (pranshugupta54) [reply clients/gui-app/src/components/chat/composer/menu/mention-preview-panel-fit.ts]: Not valid — the comments match the installed sources. `shift` uses the opposite axis convention from `flip`/`offset`: its `mainAxis` is the axis *along* the placement side, not the side axis.

`@floating-ui/utils@0.2.12`, `floating-ui.utils.mjs`:

```js
function getSideAxis(placement) {
  const firstChar = placement[0];
  return firstChar === 't' || firstChar === 'b' ? 'y' : 'x';
}
```

So `right-start` has side axis `x`.

`@floating-ui/core@1.8.0`, `floating-ui.core.mjs`, inside `shift`'s `fn`:

```js
const crossAxis = getSideAxis(placement);
const mainAxis = getOppositeAxis(crossAxis);
```

For `right-start` that is `crossAxis = 'x'`, `mainAxis = 'y'`. `shift({ mainAxis: false, crossAxis: true })` therefore clamps **x** (horizontal shifting on) and leaves y unclamped — the reverse of the review's reading.

`@floating-ui/core@1.8.0`, `floating-ui.core.mjs`, inside `size`'s `fn`:

```js
const maximumClippingWidth = width - overflow.left - overflow.right;
...
if (shiftData != null && shiftData.enabled.x) {
  availableWidth = maximumClippingWidth;
}
```

With horizontal shifting enabled, `availableWidth` is the whole clipping width less padding, not the room on the selected side.

Checked against the real middleware with a stubbed floating-ui platform: 800px-wide clipping rect, anchor row at x 300–700, 352px panel → result `left-start` at x=8 (covering the row) with `size` reporting `availableWidth` 784.

The last point is already covered: the `composer-menu.tsx` comment sits on the `isMobile ? null : <MentionPreviewPanel …>` gate and ends "the panel simply does not exist below `md`".

No code change.

File: clients/gui-app/src/components/chat/composer/menu/composer-menu-middleware.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+/**
+ * The edges as they stand right now. Read on every positioning pass: the
+ * keyboard moves them while a menu is open, and `subscribeNativeKeyboardState`
+ * is the signal that it has.
+ */
+export function readComposerMenuReservedEdges(): ComposerMenuReservedEdges {
+  return {
+    topPx: readSafeAreaInsets().top,
+    bottomPx: readNativeKeyboardInsetPx(),
```
[the remark this answers is not in the document]
Comment (pranshugupta54) [reply clients/gui-app/src/components/chat/composer/menu/composer-menu-middleware.ts]: Fixed in 8c64efd70.

`readComposerMenuReservedEdges()` now reserves all four safe-area insets. The bottom edge is `Math.max(insets.bottom, readNativeKeyboardInsetPx())`, the larger rather than the sum, because the bridge publishes the plugin's `keyboardHeight` unmodified and that frame already covers the home-indicator strip. `composerMenuViewportPadding` adds the 8px gutter to each edge. The menu and the preview panel both position through it, so the fix applies to both.

New tests in `composer-menu-middleware.test.ts`, which failed before the change:
- Landscape 844×390 with 59px side insets: a menu at a caret near the right edge ends at or before 844 − 59 − 8, and one at the left edge lands at x = 67.
- The reader reports 47/34/59/59 with the keyboard closed.
- The reader reports nothing reserved when no inset is published.

Two cases guard against regressions:
- Safe bottom 34 with a 336 keyboard reserves 336, not 370. Swapping `max` for a sum fails it.
- Nothing reserved keeps the plain 8px at the sides.
